moveit_core/robot_state/include/moveit/robot_state/attached_body.h

```diff
@@ -110,6 +111,63 @@ class AttachedBody
     return attach_trans_;
   }
 
+  /** \brief Get the fixed transform to a subframe on this body (not a transform to a visual or collision shape).
+   * The frame_name needs to have the object's name prepended (e.g. "screwdriver/tip" returns true if the object's
+   * name is "screwdriver"). Returns an identity transform if frame is not found. 
+   * Use hasSubframeTransform to check if it exists.
+  */
+  const Eigen::Isometry3d& getSubframeTransform(const std::string& frame_name) const
+  {
+    // Check if the frame_name has the object's name prepended
+    int prefix_length = id_.length() + 1;
+    try 
+    {
+      auto i = subframe_poses_.find(frame_name.substr(prefix_length));
+      if (i != subframe_poses_.end())
+        return i->second;
+      throw std::runtime_error("A subframe with name " + frame_name + " was requested from object " + id_ + " but it did not exist.");
+    }
+    catch (std::exception exc) {;}
+  }
```

Comment:
Won't you have weird return error issues if you do catch a `std::exception`? Why include the try-catch in this case?

Reply:
Because the attached body's id can be longer than the frame_name, so the substring would index out of range.